### ISSUE TITLE
Strain - Instructions and tests

### ISSUE THREAD

**Issue by 552020** [exercism/elm#484]
Regarding the [Strain ](https://exercism.org/tracks/elm/exercises/strain/edit)challenge. 

I'm not sure if I understand the design of the exercise properly. But I tried to figure out how one could solve this challenge without looking at the test. When I say that I'm not sure if I understand the design of the exercise properly I mean that I'm not sure if you are supposed to look at the test to solve the challenge. 

Instructions are as follows: 

`"Given a collection and a predicate on the collection's elements, keep returns a new collection containing those elements where the predicate is true, while discard returns a new collection containing those elements where the predicate is false."` 

It is not specified at all which predicates are applicable and there are infinite of them I would say. 

A predicate in natural language is given as an example: 

`For example, given the collection of numbers:

- 1, 2, 3, 4, 5

And the predicate:

- is the number even?"is the number even?"`

The "predicates" which are evaluated are:
- `even`, i.e. being even
- `odd`, i.e. being odd
- `isFirstLetter`, i.e. being the first letter of a string
- `lessThanTen`, i.e. being less than ten.

My question is if it's a bug or a feature of this challenge to be explained that way. 

I wonder also if it would be possible to write the challenge in a way that the use of List.filter or List.partition or eventually any list functions would let the test fail. I'm not sure if an self imposed costraint makes sense when the challenge is evalutated through a text suite with doesn't check for it.

**Comment by mpizenberg** [comment]
It's ok too look at the tests, though most often one would just implement the function and look at the failing tests that will be displayed in exercism UI.

You're right that the "filter" restriction cannot be enforced in the tests results. However, since a few weeks ago, and thanks to @jiegillet mainly, we have a piece of infrastructure called an analyzer in the elm track. And strain is the first exercise to have such analyzer rules. So once you've submitted your solution with `filter` in then, you will see on the exercise page something like this the screenshot below.

![image](https://user-images.githubusercontent.com/2905865/161755331-a11aa2b3-0617-444d-9307-cd022d03bd8c.png)

And if you click on the red robot thingy, the analyzer will detect "filter" usage and ask you to change.

![image](https://user-images.githubusercontent.com/2905865/161755545-aa891cc4-061a-4ae8-9274-8153546f03c0.png)

**Comment by mpizenberg** [comment]
Maybe we want to add in the exercise descriptions something like "The elm analyzer will check your code once it's submit, don't forget to look at the analyzer feedback."

**Comment by jiegillet** [comment]
> Maybe we want to add in the exercise descriptions something like "The elm analyzer will check your code once it's submit, don't forget to look at the analyzer feedback."

I don't think this is necessary. As we roll out more rules (especially common ones), it will become expected.

**Comment by mpizenberg** [comment]
Alright, let's see how the analyzer deployment of common rules unfolds soon then :)

https://github.com/exercism/elm-analyzer/pull/29
